zerver/lib/onboarding.py

```diff
@@ -131,6 +134,35 @@ def send_initial_direct_message(user: UserProfile) -> None:
         disable_external_notifications=True,
     )
 
+    custom_welcome_bot_message_string = ""
+    custom_welcome_bot_message_text = ""
+    # Send additional welcome bot message configured by administrators.
+    if custom_welcome_bot_message is not None:
+        custom_welcome_bot_message_text = custom_welcome_bot_message.strip()
+    elif (
+        user.realm.custom_welcome_bot_message_enabled
+        and user.realm.custom_welcome_bot_message is not None
+    ):
+        custom_welcome_bot_message_text = user.realm.custom_welcome_bot_message.strip()
+
+    if custom_welcome_bot_message_text:
+        custom_welcome_bot_message_string = (
+            _(
```

Comment:
This won't get translated properly as it is outside `with override_language(user.default_language):` block.

Also, recently we have started using `remove_single_newlines()`, we should use that here.

Reply:
Thank you for the suggestion. However, the quote\n{realm_custom_welcome_bot_message}\n block requires the newlines to maintain the correct formatting. Using remove_single_newlines here would disrupt the intended structure of the message.